robot-db/src/main/java/repositories/BookInput.java

```diff
@@ -0,0 +1,62 @@
+package repositories;
+
+import dao.BookDAO;
+import domain.Author;
+import domain.Book;
+import domain.Category;
+import domain.PrintHouse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Class responsible for Inserting Parsed Book into Database
+ */
+@Repository
+@Slf4j
+public class BookInput {
+
+    @Autowired
+    private AuthorInput authorInput;
+
+    @Autowired
+    private CategoryInput categoryInput;
+
+    @Autowired
+    private PrintHouseInput printHouseInput;
+
+    @Autowired
+    private BookDAO bookDAO;
+
+    /**
+     * Saves the book in BookDAO object.
+     *
+     * @param parsedBook - book obtained from Parser.
+     */
+    public void save (ParsedBook parsedBook) {
+        log.info("staring operations on database");
+        Category category = categoryInput.saveCategory(parsedBook.getCategory());
```

Comment:
Try reading that out loud and then tell me it sounds well to your ears, that SAVING a category to DB returns you a category to operate, and that this is a category INPUT operation. What are your DAO doing?